packages/discussions/src/utils/posts.ts

```diff
@@ -84,3 +93,33 @@ export function isDiscussable(subject: IGroup | IItem | IHubContent) {
   // TODO: implement
   return true;
 }
+
+/**
+ * Determines if the given user has sufficient privileges to modify the given post
+ * @param post An IPost object
+ * @param channel An IChannel object
+ * @param user An IUser object
+ * @returns true if the user can modify the post
+ */
+export function canModifyPost(
+  post: IPost,
+  channel: IChannel,
+  user: IUser
+): boolean {
+  return post.creator === user.username || canModifyChannel(channel, user);
```

Comment:
Why do we need to check if the user can modifyChannel for them to modify a post?

Reply:
Actually I guess we do need it since canModifyChannel says `Utility to determine whether User can modify channel settings and posts belonging to Channel`

Reply:
@jersteve 

This was taken directly from the API logic. Once this PR merges, we can update lines [#168](https://devtopia.esri.com/dc/hub-discussions/blob/main/src/posts/posts.service.ts#L168) and [#284](https://devtopia.esri.com/dc/hub-discussions/blob/main/src/posts/posts.service.ts#L284) to use these new utility methods. The frontend will be utilizing these new utils too. This will help keep things in sync between the frontend and backend as that logic changes.